sys/netgraph/ng_pipe.c

```diff
@@ -658,7 +660,8 @@ ngp_rcvdata(hook_p hook, item_p item)
 			break;
 	if (ngp_f == NULL) {
 		ngp_f = uma_zalloc(ngp_zone, M_NOWAIT);
-		KASSERT(ngp_h != NULL, ("ngp_h zalloc failed (2)"));
+		if (ngp_f == NULL)
+			return (ENOMEM);
```

Comment:
Here you're leaking ngp_h (and maybe other things).

Reply:
Good catch!